### DISCUSSION TITLE
[QUESTION] Double checking if the max number of files is still 2 billons

### DISCUSSION THREAD

**Discussion by ptempier** [moosefs/moosefs#617] [Q&A]
Hello

I found this number in the doc, i am wondering if this number was still valid.
Nowaday, such number doesn't seems that high.
(My max is 1 billion inode used by a single group on a single server, its not ..yet... on moosefs)

I am wondering if 2Bn is the max number of inodes ( files+folder+link ) or files only.

If snapshot/archives, when used also decrease that count.

I didn t found this info in the doc, but is it possible to somehow increase this limit ?

**Comment by chogata** [comment]
Yes, we still have the limit, this is because we still use 32bit inodes, this is because converting to 64bits will cause a LOT of problems with 32bit OSes, which are still widely used.

Snapshots create new inodes for files and directories, so yes, they will "use up" the limit.

**Comment by everanium** [comment]
Hello, I have a question. Why not create a separate mode that would use 64-bit inodes during cluster initialization to remove this limitation? Some physical servers have a large amount of RAM (several terabytes), meaning the metadata will fit in the master server's memory, even if there are more than 2 billion files in the cluster file system.
